hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java

```diff
@@ -86,9 +95,10 @@ protected IdentifiedDataSerializable getConfig() {
         config.setValueType(parameters.valueType);
         config.setWanReplicationRef(parameters.wanReplicationRef);
         config.setWriteThrough(parameters.writeThrough);
-        if (parameters.isMerkleTreeConfigExists) {
+        if (parameters.isMerkleTreeConfigExists && parameters.merkleTreeConfig != null) {
             config.setMerkleTreeConfig(parameters.merkleTreeConfig);
         }
+        config.setDataPersistenceConfig(parameters.dataPersistenceConfig);
```

Comment:
Should be conditionally set if `parameters.dataPersistenceConfig != null`. Consider the case of an older (eg 5.1) client sending an `AddCacheConfig` message: `dataPersistenceConfig` will be `null` and `config.setDataPersistenceConfig` will throw `NullPointerException`. See similar issue reported in #21732 for map config.

Reply:
Thanks for catching it. I'll take care of the add map config too.